app/controllers/playlists_controller.rb

```diff
@@ -16,6 +16,8 @@ def new
 
   def create
     @playlist = Playlist.new(playlist_params)
+    @user = current_user
+    @playlist.user = @user
```

Comment:
```suggestion
    @playlist = Playlist.new(playlist_params)
```

et plus bas dans ton `playlist_params`:

```ruby
def playlist_params
  params.require(:playlist).permit(:name, :genre, :danceability, :energy, :spotify_id).merge(user: current_user)
end
```

De manière général c'est mieux de n'avoir qu'une seule variable d'instance par action. Ici en plus `@user` est accessible avec `current_user`, ce qui est plus clair si tu dois le manipuler dans la vue.